parser.js

```diff
@@ -184,9 +184,35 @@ class Extractor {
         return {'name': name, 'scope': scope, 'type': type, 'visibility': visibility}
     }
 
+    _parseCommaToBar(params) {
+        let refactorString = '';
+        let foundArrow = false;
```

Comment:
Use a int with `arrowDeep = 0` instead of a boolean.

At each `<` increment `arrowDeep`, and at each `>` decrement it.

This will allow multiple generics type (test case: `test: int, crazyMap: HashMap<List<Integer>, HashMap<String, List<List<Double>>>> , test: str`).

Also, nice work!

Reply:
Oh nice idea! I will do this.